src/modules/Hooks.ts

```diff
@@ -71,6 +75,12 @@ export type Handlers = {
 	[K in HookName]: HookHandler<K>[];
 };
 
+export type HookInitOptions = {
```

Comment:
It looks to me like this could be simplified:

```js
export type HookInitOptions = {
	[K in HookName as HookName | HookNameWithModifier]: HookHandler<K>;
};
```

Reply:
Updated with ``[K in HookName as K | `${K}.${HookModifier}`]: HookHandler<K>`` 🍻